### DISCUSSION TITLE
Schema Validation Using Factory Pattern

### DISCUSSION THREAD

**Discussion by raunakdeshmukh** [pydantic/pydantic#2870] [Question]
Firstly, this is my first implementation of Pydantic, so any help is appreciated

I am trying to build a FastAPI and doing Input Validation using Pydantic

The model I use depends on the input values

For the sake of an example, 

this is one of my endpoint:

```
@endpoint.router.post("/InputApi")
def json_example(example_model: Vehicle):
```

The basic schema of the Vehicle is:

```
class Vehicle(BaseModel):
    """Vehicle schema"""
    Type: int
    Data = Any
```

The 'Type' in the above model decides what model the List will contain

So I have two more models:

```
class Truck(BaseModel):
    """Truck schema"""
    Height: int
    Weight = int
```

```
class Car(BaseModel):
    """Car schema"""
    Occupancy: int
    Price: int
```

I have a schema factory which will basically be called from the Vehicle class, 

```
import Car, Truck

class VehicleSchemaFactory:

        factory = {
                            1: Truck,
                            2: Car
                         }
```

What could be the right way to do something close to this in the vehicle model

```
import VehicleSchemaFactory
class Vehicle(BaseModel):
    """Vehicle schema"""
    Type: int
    Data = Any

    @validator('Data')
     def validate_data_schema(cls, v, values)
             if Type in values:
                     return VehicleSchemaFactory.get(Type)
```

I am sure this is not the right usage of the validator operator, but I am just trying to explain the usecase

**Comment by PrettyWood** [comment]
Hi @raunakdeshmukh 
If I understood correctly, please see https://github.com/samuelcolvin/pydantic/issues/2177#issuecomment-739578307 for an example
